packages/teams-js/src/internal/validDomains.ts

```diff
@@ -0,0 +1,104 @@
+import { GlobalVars } from '../internal/globalVars';
+import { validDomainsCdnEndpoint, validOriginsFallback } from './constants';
+import { getLogger } from './telemetry';
+import { isValidHttpsURL } from './utils';
+
+let validOrigins: string[] = [];
+export async function prefetchDomainsFromCDN(): Promise<void> {
+  if (fetch) {
+    await fetch(validDomainsCdnEndpoint);
+  }
+}
+
+async function retrieveDomainsFromCDNAndStore(): Promise<string[]> {
```

Comment:
>retrieveDomainsFromCDNAndStore [](http://example.com/codeflow?start=15&length=30)

This name is descriptive, which is GREAT. I think one thing that could be clearer is that it's not just retrieving/storing the domains, it's also returning them.

Perhaps something like `getValidDomains` might be better since, to the caller, all they are looking to do is get the valid domains. The fact that sometimes they are cached and sometimes it goes to the CDN etc. seems like a implementation detail?

I know, I know, it feels like I sometimes say "descriptive names are great" and here I am suggesting that maybe this is too descriptive. 😅

Reply:
Just want to call out that this name is this way because I asked Jay to change it in a previous iteration. I'm fine with Aaron's name (especially since this now returns those domains and didn't previously).

Reply:
Yes, @TrevorJoelHarris and I aren't fighting. 😅 I think the implementation of what the function did changed over time so the potential naming suggestions also changed.